backend/drm/renderer.c

```diff
@@ -277,22 +263,38 @@ bool drm_plane_init_surface(struct wlr_drm_plane *plane,
 		drm_format = wlr_drm_format_dup(&format_no_modifiers);
 	}
 
+	struct wlr_drm_format *drm_format_linear =
+		calloc(1, sizeof(struct wlr_drm_format) + sizeof(uint64_t));
+	if (drm_format_linear == NULL) {
+		return false;
+	}
+	drm_format_linear->format = drm_format->format;
+	drm_format_linear->len = 1;
+	drm_format_linear->cap = 1;
+	drm_format_linear->modifiers[0] = DRM_FORMAT_MOD_LINEAR;
+
+	if (force_linear) {
+		free(drm_format);
+		drm_format = wlr_drm_format_dup(drm_format_linear);
+	}
+
 	drm_plane_finish_surface(plane);
 
 	bool ok = true;
 	if (!drm->parent) {
-		ok = init_drm_surface(&plane->surf, &drm->renderer, width, height,
-			drm_format, flags | GBM_BO_USE_SCANOUT);
+		ok = init_drm_surface(&plane->surf, &drm->renderer,
+			width, height, drm_format);
 	} else {
 		ok = init_drm_surface(&plane->surf, &drm->parent->renderer,
-			width, height, drm_format, flags | GBM_BO_USE_LINEAR);
+			width, height, drm_format_linear);
 		if (ok && !init_drm_surface(&plane->mgpu_surf, &drm->renderer,
```

Comment:
A question, unrelated to the PR: what is `mgpu_surf` aka what does purpose does this second attempt at init_drm_surface achieve?

Reply:
On multi-GPU setups, we need two surfaces: one on the primary GPU for rendering operations, and one on the secondary GPU to copy over the final buffer for scan-out. The code initializes both surfaces here.